Source/WebCore/dom/Document.cpp

```diff
@@ -11076,6 +11077,28 @@ void Document::securityOriginDidChange()
     m_permissionsPolicy = nullptr;
 }
 
+#if ENABLE(CONTENT_EXTENSIONS)
+
+ResourceMonitor* Document::resourceMonitor()
+{
+    return m_resourceMonitor.get();
+}
+
+void Document::setResourceMonitor(RefPtr<ResourceMonitor>&& resourceMonitor)
+{
+    m_resourceMonitor = WTFMove(resourceMonitor);
+}
+
+ResourceMonitor* Document::parentResourceMonitor()
```

Comment:
Don't we want the top-most ResourceMonitor? What if there is frame nesting like so?
```
A
  B - Eligible
    C
      D - Shouldn't this look up B's monitor?
```

Reply:
When frame is created inside eligible frame, that frame is also eligible in the constructor and no check is done further more.

Reply:
Ok, I talked to Basuke about this, and he points out that B, C, and D should always have a resource monitor (only the root doesn't have one). When you call `ResourceMonitor::addNetworkUsage` on D, it reports network usage up the tree all the way up to A unconditionally.

Reply:
And resource monitor is always created except for the main frame and no chance if C doesn't have resource monitor.